proxy/http2/HTTP2.cc

```diff
@@ -448,7 +448,7 @@ convert_from_2_to_1_1_header(HTTPHdr *headers)
 
     // Combine Cookie headers ([RFC 7540] 8.1.2.5.)
     field = headers->field_find(MIME_FIELD_COOKIE, MIME_LEN_COOKIE);
-    if (field) {
```

Comment:
This shouldn't be necessary since you are going to check all the header below.  You would be validating the cookies two times.